GVFS/GVFS/CommandLine/HealthVerb.cs

```diff
@@ -0,0 +1,227 @@
+            HelpText = "Run the statistics tool on a specific directory")]
+        public string Directory { get; set; }
+
+        protected override string VerbName
+        {
+            get { return HealthVerbName; }
+        }
+
+        protected override void Execute(GVFSEnlistment enlistment)
+        {
+            // The path to the root of git's tree
+            string sourceRoot = enlistment.WorkingDirectoryRoot;
+
+            if (this.Directory == null)
+            {
+                this.Directory = string.Empty;
+            }
+            else if (this.Directory.Equals("."))
+            {
+                this.Directory = Environment.CurrentDirectory.Substring(sourceRoot.Length);
+            }
+
+            // Get all of the data needed to calculate statistics then pass them into GVFSEnlistmentStatistics
+            List<string> modifiedPathsFileList = new List<string>();
+            List<string> modifiedPathsFolderList = new List<string>();
+            List<string> placeholderFilePathList = new List<string>();
+            List<string> placeholderFolderPathList = new List<string>();
+
+            this.GetPlaceholdersFromDatabase(enlistment, out List<IPlaceholderData> filePlaceholders, out List<IPlaceholderData> folderPlaceholders);
+            foreach (IPlaceholderData placeholderData in filePlaceholders)
+            {
+                placeholderFilePathList.Add(placeholderData.Path);
+            }
+
+            foreach (IPlaceholderData placeholderData in folderPlaceholders)
+            {
+                placeholderFolderPathList.Add(placeholderData.Path);
+            }
+
+            foreach (string path in this.GetModifiedPathsFromPipe(enlistment))
+            {
+                if (path.Last() == GVFSConstants.GitPathSeparator)
+                {
+                    path.TrimEnd('/');
+                    modifiedPathsFolderList.Add(path);
+                }
+                else
+                {
+                    modifiedPathsFileList.Add(path);
+                }
+            }
+
+            List<string> gitPathsList = this.GetPathsFromGitIndex(enlistment);
+
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData pathData = new GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData(
+                gitPathsList,
+                placeholderFolderPathList,
+                placeholderFilePathList,
+                modifiedPathsFolderList,
+                modifiedPathsFileList);
+            GVFSEnlistmentHealthCalculator enlistmentHealthCalculator = new GVFSEnlistmentHealthCalculator(pathData);
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData = enlistmentHealthCalculator.CalculateStatistics(this.Directory, this.DirectoryDisplayCount);
+
+            this.PrintOutput(enlistmentHealthCalculator, enlistmentHealthData);
+        }
+
+        private void PrintOutput(
+            GVFSEnlistmentHealthCalculator enlistmentHealthCalculator,

```

Comment:
Do we need the `GVFSEnlistmentHealthCalculater` object in order to print statistics?

Reply:
Restructured the `GetPlaceholderPercentage` and `GetModifiedPathsPercentage` methods to be properties within the `GVFSEnlistmentHealthData` which compute on the fly.